python/pyspark/sql/pandas/types.py

```diff
@@ -294,8 +303,10 @@ def _check_series_convert_timestamps_localize(
     from pandas.api.types import is_datetime64tz_dtype, is_datetime64_dtype
     from_tz = from_timezone or _get_local_timezone()
     to_tz = to_timezone or _get_local_timezone()
-    # TODO: handle nested timestamps, such as ArrayType(TimestampType())?
-    if is_datetime64tz_dtype(s.dtype):
+    if datatype == ArrayType.__name__:
```

Comment:
I don't think it's a good way to check for data types by using the `__name__` attribute, isn't there another way to check that the data type is an ArrayType?

Reply:
Will change it